File: content/problems/2270-number-of-ways-to-split-array/solutions/rolling-prefix-comparison/Solution.java
class Solution {
    public int waysToSplitArray(int[] nums){
        long total=0,prefix=0;
        for(int value:nums)total+=value;
        int answer=0;
        for(int i=0;i+1<nums.length;i++){
            prefix+=nums[i];
            if(prefix>=total-prefix)answer++;
        }
        return answer;
    }
}

File: content/problems/2270-number-of-ways-to-split-array/solutions/rolling-prefix-comparison/Solution.kt
class Solution {
    fun waysToSplitArray(nums:IntArray):Int{
        val total=nums.sumOf{
            it.toLong()
        };
        var prefix=0L;
        var answer=0;
        for(i in 0 until nums.lastIndex){
            prefix+=nums[i];
            if(prefix>=total-prefix)answer++
        };
        return answer
    }
}
